src/lib/scheduler/node_queue_scheduler.hpp

```diff
@@ -129,7 +122,7 @@ class NodeQueueScheduler : public AbstractScheduler {
   std::vector<std::shared_ptr<Worker>> _workers;
 
   std::atomic_bool _active{false};
-  std::atomic_uint64_t _active_worker_count{0};
+  std::atomic_int64_t _active_worker_count{0};
```

Comment:
Love how you do everything to not assign and dcrement in one line :D

Reply:
Not sure it would work here.

Reply:
https://en.cppreference.com/w/cpp/atomic/atomic/operator_arith (4)
So the following should work:
```c++
const auto remaining_workers = _active_worker_count--;
Assert(remaining_workers > 0, "...");
```

Reply:
Ah nice.

Reply:
Wouldn't solve the issue with the `uint64_t` though.

Reply:
Sure. At least when using post-decrement.
> ### Return value
>1,3) The value of the atomic variable after the modification. Formally, the result of incrementing/decrementing the value immediately preceding the effects of this function in the [modification order](https://en.cppreference.com/w/cpp/atomic/memory_order#Modification_order) of *this.
2,4) The value of the atomic variable before the modification. Formally, the value immediately preceding the effects of this function in the [modification order](https://en.cppreference.com/w/cpp/atomic/memory_order#Modification_order) of *this.

So it should be > 0 before we decrement, which is what we get by the suggestion above.

Reply:
True. 🙈